collector crash

{"level":"info","msg":"setting up collector for devices","numDevices":2,"time":"2021-06-17T18:43:12+03:00"}
{"level":"info","msg":"Listening on :9436","time":"2021-06-17T18:43:12+03:00"}
panic: label value "\xef-HP" is not valid UTF-8

goroutine 4 [running]:
github.com/prometheus/client_golang/prometheus.MustNewConstMetric(0x215f6c0, 0x2, 0x0, 0x3ff00000, 0x217a940, 0x8, 0x8, 0x527698, 0x2210940)
        /root/go/pkg/mod/github.com/prometheus/client_golang@v1.4.1/prometheus/value.go:105 +0x7c
mikrotik-exporter/collector.(*dhcpLeaseCollector).collectMetric(0x2110880, 0x2111300, 0x21f7100)
        /root/src/mikrotik-exporter-src/collector/dhcp_lease_collector.go:81 +0x5a4
mikrotik-exporter/collector.(*dhcpLeaseCollector).collect(0x2110880, 0x2111300, 0x0, 0x0)
        /root/src/mikrotik-exporter-src/collector/dhcp_lease_collector.go:42 +0x6c
mikrotik-exporter/collector.(*collector).connectAndCollect(0x2122450, 0x20fb300, 0x20fb140, 0x0, 0x0)
        /root/src/mikrotik-exporter-src/collector/collector.go:348 +0x29c
mikrotik-exporter/collector.(*collector).collectForDevice(0x2122450, 0x2126820, 0x6, 0x2126840, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /root/src/mikrotik-exporter-src/collector/collector.go:319 +0x6c
mikrotik-exporter/collector.(*collector).Collect.func1(0x2122450, 0x20fb140, 0x201a0d0, 0x2126820, 0x6, 0x2126840, 0xc, 0x0, 0x0, 0x0, ...)
        /root/src/mikrotik-exporter-src/collector/collector.go:284 +0x30
created by mikrotik-exporter/collector.(*collector).Collect
        /root/src/mikrotik-exporter-src/collector/collector.go:283 +0x1c0

--- a/collector/dhcp_lease_collector.go
+++ b/collector/dhcp_lease_collector.go
@@ -76,7 +76,8 @@ func (c *dhcpLeaseCollector) collectMetric(ctx *collectorContext, re *proto.Sent
 	server := re.Map["server"]
 	status := re.Map["status"]
 	activeaddress := re.Map["active-address"]
-	hostname := re.Map["host-name"]
+	// QuoteToASCII because of broken DHCP clients
+	hostname := strconv.QuoteToASCII(re.Map["host-name"])
 
 	metric, err := prometheus.NewConstMetric(c.descriptions, prometheus.GaugeValue, v, ctx.device.Name, ctx.device.Address, activemacaddress, server, status, strconv.FormatFloat(f, 'f', 0, 64), activeaddress, hostname)
 	if err != nil {

